Plugins/NativeCamera/Source/Android/CameraWrappers.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <android/api-level.h>
+#include <dlfcn.h>
+#include <map>
+
+// The camera API's in the NDK are only supported in API24 and higher. We
+// use dlopen and dlsym to link the camera lib at runtime only if the current
+// device supports it. In order to avoid redefining the type information the
+// below macro's forcefully import the camera header files as if the min supported
+// SDK was set to 24.
+#define __ORIG_ANDROID_API__ __ANDROID_API__
+#undef __ANDROID_API__
+#define __ANDROID_API__ 24
+namespace API24 {
+#include <camera/NdkCameraManager.h>
+    #include <camera/NdkCameraCaptureSession.h>
+    #include <camera/NdkCameraDevice.h>
+    #include <camera/NdkCameraError.h>
+    #include <camera/NdkCameraManager.h>
+    #include <camera/NdkCameraMetadata.h>
+    #include <camera/NdkCameraMetadataTags.h>
+    #include <camera/NdkCameraWindowType.h>
+    #include <camera/NdkCaptureRequest.h>
+}
+#undef __ANDROID_API__
+#define __ANDROID_API__ __ORIG_ANDROID_API__
+#undef __ORIG_ANDROID_API__
+
+// Helper macro to make calling camera NDK api's more type safe
+#define GET_CAMERA_FUNCTION(function) reinterpret_cast<decltype(&API24::function)>(Babylon::Plugins::GetCameraDynamicFunction(#function))
```

Comment:
This is an interesting way to do it. I'm not sure I like it. The more typical way is to store each function the code will call in an explicit list of function pointers.

Reply:
There are two main reasons I went with this route of loading the functions on demand:
1. If you don't end up using the functions you don't pay any cost of finding them in the library. It's probably a negligible amount of time... but I liked knowing they wouldn't get loaded if the NativeCamera wasn't used.
2. It reduces the boilerplate needed to call a new function out of the NDK Camera API. I'm hoping to do some follow up PR's to make the NativeCamera implementation a bit more robust (such as the issues called out in #1107). In doing so I'd like to be able to call other functions in the Camera API as easily as possible. With this current route the macro makes that very easy. Pre-defining the functions isn't much extra work... but it is a little.

As long as you're okay with it in it's current form I'd like to leave it that way and then change it later if it becomes an issue.

Reply:
> If you don't end up using the functions you don't pay any cost of finding them in the library. It's probably a negligible amount of time... but I liked knowing they wouldn't get loaded if the NativeCamera wasn't used.

This is both a pro and a con. The con part is that you won't know about the missing function until you hit the code path that uses it which can cause bugs to slip through. It's certainly not a deal breaker.